buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchDistribution.java

```diff
@@ -107,17 +107,18 @@ public String toString() {
     private final Property<Flavor> flavor;
     private final Property<Boolean> bundledJdk;
 
-    ElasticsearchDistribution(String name, Project project) {
+    ElasticsearchDistribution(String name, ObjectFactory objectFactory, Configuration fileConfiguration,
```

Comment:
FWIW, we can actually use Gradle's service injection here since it's actually doing the instantiation. Just add a getter like so and we can avoid having to explicitly pass this to the constructor altogether.

```
@Inject
public ObjectFactory getObjectFactory() {
    throw new UnsupportedOperationException();
}
```